pkg/reaper/service.go

```diff
@@ -40,7 +49,7 @@ func NewService(key types.NamespacedName, reaper *api.Reaper) *corev1.Service {
 					StrVal: "admin",
 				},
 			}},
-			Selector: labels,
+			Selector: getConstantLabels(reaper),
```

Comment:
I think this is changing the semantics slightly to limit the selector labels to just the constant ones, right? I was a bit wary initially but I saw that you added more assertions in `TestNewService` and the results seem OK, so 👍 for the change.